src/main/kotlin/com/openlattice/mechanic/upgrades/CleanOutOrgMembersAndRoles.kt

```diff
@@ -0,0 +1,55 @@
+package com.openlattice.mechanic.upgrades
+
+import com.openlattice.authorization.Ace
+import com.openlattice.authorization.Acl
+import com.openlattice.authorization.AuthorizationManager
+import com.openlattice.authorization.Permission
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.organizations.roles.SecurePrincipalsManager
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class CleanOutOrgMembersAndRoles(
+        private val toolbox: Toolbox,
+        private val principalsManager: SecurePrincipalsManager,
+        private val authorizationManager: AuthorizationManager
+) : Upgrade {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(CleanOutOrgMembersAndRoles::class.java)
+        private val MEMBER_PERMISSIONS = EnumSet.of(Permission.READ)
+    }
+
+    override fun upgrade(): Boolean {
+        val organizationsMapstore = HazelcastMap.ORGANIZATIONS.getMap(toolbox.hazelcast)
+
+        val organizations = organizationsMapstore.toMap()
+        val principals = HazelcastMap.PRINCIPALS.getMap(toolbox.hazelcast).values.associate { it.principal to it.aclKey }
+
+        organizations.values.forEach {
+            logger.info("About to clean out members for org {} [{}]", it.title, it.id)
+
+            val memberAclKeys = it.members.mapNotNull { p -> principals[p] }.toSet()
+            principalsManager.addPrincipalToPrincipals(it.getAclKey(), memberAclKeys)
+            authorizationManager.addPermissions(listOf(
+                    Acl(
+                            it.getAclKey(),
+                            it.members.map { p -> Ace(p, MEMBER_PERMISSIONS) }
+                    )
+            ))
+
+            it.members.clear()
+            it.roles.clear()
+            organizationsMapstore[it.id] = it
```

Comment:
Ok, so... just checking my understanding... `it.members.clear()` and `it.roles.clear()` will nuke members and roles, and `it.members.mapNotNull` will change "members" to be a set of acl keys. After this runs, is this what the org object will look like?

```
{
  id: org_uuid,
  title: "the org",
  description: "the org",
  ...
  members: [
    [member_securable_principal_uuid_1],
    [member_securable_principal_uuid_2],
    ...
  ],
  roles: []
}
```